mitiq/qse/tests/test_qse.py

```diff
@@ -109,44 +109,27 @@ def test_mitigate_executor_batched():
         code_hamiltonian,
         observable,
     )
-    unmitigated_value = observable.expectation(
-        qc, execute_with_depolarized_noise
-    )
-
-    batched_mitigated_values = batched_mitigated_executor([qc] * 3)
-    assert all(
-        abs(mitigated_value.real - 0.5) < abs(unmitigated_value.real - 0.5)
-        for mitigated_value in batched_mitigated_values
-    )
 
+    num_circuits = 3
+    circuits = [qc] * num_circuits
+    batched_mitigated_executor(circuits)
 
-def test_qse_decorator():
-    qc = prepare_logical_0_state_for_5_1_3_code()
-    (
+    mock_execute_with_qse.assert_called_with(
+        circuits[0],
+        batched_execute_with_depolarized_noise,
         check_operators,
         code_hamiltonian,
-    ) = get_5_1_3_code_check_operators_and_code_hamiltonian()
-    observable = get_observable_in_code_space(PauliString("ZZZZZ"))
-
-    @qse_decorator(
-        check_operators=check_operators,
-        code_hamiltonian=code_hamiltonian,
-        observable=observable,
+        observable,
+        {},
     )
-    def decorated_executor(circuit: QPROGRAM) -> np.ndarray:
-        return compute_density_matrix(
-            convert_to_mitiq(circuit)[0],
-            noise_model_function=cirq.depolarize,
-            noise_level=(0.01,),
-        )
+    assert mock_execute_with_qse.call_count == num_circuits
```

Comment:
Do you know how `mock_execute_with_qse.call_count` "knows" that it should return `num_circuits`?
All seems to work. I am just curious how `unittest.mock.patch` is able to mock functions without explicitly setting return values.

Reply:
I was also surprised you don't have to set a return value, but it's nice not having to set some random value for it since it's meaningless (in this instance).

As for how it actually works, [`unittest.mock.patch`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch) creates a [`MagicMock`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.MagicMock) instance for the `execute_with_qse` function. Under the hood my guess is that the `MagicMock` class looks something like this:

```py
class MagicMock:
    def __init__(self):
        self.call_count = 0
    
    def __call__(self):
        self.call_count += 1
```